src/analyses/README.md

```diff
@@ -32,7 +32,10 @@ To be documented.
 
 \subsection analyses-constant-propagation Constant propagation (constant_propagator_ait)
 
-To be documented.
+A simple, unsound constant propagator. Replaces RHS symbol expressions (variable
+reads) with their values when they appear to have a unique value at a particular
+program point. Unsound with respect to pointer operations on the left-hand side
```

Comment:
Is that genuinely the case? It's probably unsound for concurrent code, but I'm not sure about pointers.

Reply:
Yes: https://github.com/diffblue/cbmc/blob/develop/src/analyses/constant_propagator.cpp#L38 (assign with anything but a symbol on the LHS is a no-op)

Reply:
Ouch. I wasn't quite aware how incomplete it was.